woocommerce/woocommerce-android
# Enable Sentry Profiling

- feat: disable auto-activity traces
- feat: enable Sentry profiling
- feat: increase performance monitoring sampling



Closes: #9178


### Description

This PR enables Sentry Profiles recording.

Firstly, review & merge https://github.com/Automattic/Automattic-Tracks-Android/pull/178

### Testing instructions


Apply PATCH

<details><summary>PATCH</summary>

```PATCH
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/SpecifyPerformanceMonitoringConfig.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/SpecifyPerformanceMonitoringConfig.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/SpecifyPerformanceMonitoringConfig.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/SpecifyPerformanceMonitoringConfig.kt	(revision ea516dabb881ab221dd5ddd1fe416ca1f3049927)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/SpecifyPerformanceMonitoringConfig.kt	(date 1685704239138)
@@ -18,12 +18,12 @@
     operator fun invoke(): PerformanceMonitoringConfig {
         val userEnabled = analyticsWrapper.sendUsageStats
 
-        return if (!userEnabled || buildConfig.debug) {
+        return if (!userEnabled) {
             PerformanceMonitoringConfig.Disabled
         } else {
             PerformanceMonitoringConfig.Enabled(
-                sampleRate = PERFORMANCE_MONITORING_SAMPLE_RATE,
-                profilesSampleRate = PROFILING_SAMPLE_RATE
+                sampleRate = 1.0,
+                profilesSampleRate = 1.0
             )
         }
     }
```
</details>

1. Run the app
2. Open Order Details
3. See that there's a new OrderDetails transaction in Sentry dashboard with attached Profile.


### Images/gif


<img width="1118" alt="image" src="https://github.com/woocommerce/woocommerce-android/assets/5845095/33efd582-e1fd-4d4a-a5b8-2b7f418a762c">
